src/configs.ts

```diff
@@ -83,13 +140,19 @@ export async function resolveConfigPath(options: ResolveConfigPathOptions) {
  */
 export async function readConfig(
   options: ReadConfigOptions,
-): Promise<{ configs: FlatConfigItem[], payload: Payload, dependencies: string[] }> {
+): Promise<ConfigPathLegacyError | ConfigPathError | { configs: FlatConfigItem[], payload: Payload, dependencies: string[] }> {
```

Comment:
Why don't we use throw instead?

Reply:
@antfu As throws are not typed in TypeScript and can be intertwined with other kinds of exceptions

Reply:
I think we could throw and use `if (error instanceof ConfigPathLegacyError)` to do the type guard

Reply:
We could, but not sure it would make the code simpler :)

Reply:
I don't think it would make it more complicated either - but I think the point is that errors are supposed to be thrown, and `loadConfig` by it's name should returns the config

Reply:
Sorry that I never got to fixing this 🙈

Speaking on this topic, ran into this standard proposal the other day: https://github.com/arthurfiorette/proposal-safe-assignment-operator

Reply:
No worries at all.

Yeah I saw it. Would love to see it landed someday!